lbz/handler_base.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import Generic, TypeVar
+
+from lbz.misc import deprecated
+
+T = TypeVar("T")
+
+
+class HandlerBase(Generic[T]):
+    def react(self) -> T:
+        self.pre_handle()
+        response = self.handle()
+        self.post_handle()
+        return response
+
+    @deprecated(message="Please use react() for full request flow")
+    def __call__(self) -> T:
+        return self.react()
+
+    @abstractmethod
+    def handle(self) -> T:
+        pass
+
+    def pre_handle(self) -> None:
```

Comment:
It looks like both `pre_handle()` and `post_handle()` are not tested anywhere 😐 

I think you should write additional tests for this base class, thanks to that, we won't have to repeat them for each class that inherits from this code.